### ISSUE TITLE
Pylint C0103 (invalid-name) contradiction when module level names matches will class level names

### ISSUE THREAD

**Issue by RomanValov** [pylint-dev/pylint#10719]
### Bug description

```python
class Theme:
    INPUT = ">>> "

INPUT = Theme()
Input = Theme()

OUTPUT = Theme()
Output = Theme()
```

With recent updates of pylint it starts to complain about `Input` identifier in the code. But changing `Input` to `INPUT` produces another complaint. However if I delete `INPUT` field from class `Theme`, pylint no longer complains about module-level `INPUT`.

### Configuration

```ini

```

### Command used

```shell
pylint --disable=all --enable=C0103 ./theme.py
```

### Pylint output

```python
************* Module theme
theme.py:5:0: C0103: Variable name "INPUT" doesn't conform to snake_case naming style (invalid-name)
theme.py:6:0: C0103: Constant name "Input" doesn't conform to UPPER_CASE naming style (invalid-name)
theme.py:9:0: C0103: Constant name "Output" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 3.33/10, +1.67)
```

### Expected behavior

Module-level identifiers `INPUT` and `OUTPUT` are treated equally interdependently of class-level identifiers in the same module

### Pylint version

```shell
pylint 4.0.2
astroid 4.0.1
Python 3.12.3 (main, Aug 14 2025, 17:47:21) [GCC 13.3.0]
```

### Additional dependencies

```python

```

**Comment by joao-faria-dev** [comment]
Hey @RomanValov, i just raised a PR related to this issue. Hope it helps :)
